src/cal/mod.rs

```diff
@@ -86,7 +198,12 @@ impl CmpEvent {
 
 impl Ord for CmpEvent {
     fn cmp(&self, other: &CmpEvent) -> Ordering {
-        self.event.date.cmp(&other.event.date)
+        let comp = self.event.date.cmp(&other.event.date);
+        if comp == Ordering::Equal {
+            self.event.duration.cmp(&other.event.duration)
+        } else {
+            comp
+        }
```

Comment:
With https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then this becomes

``` Rust
self.event.date.cmp(&other.event.date).then(self.event.duration.cmp(&other.event.duration))
```

Also, was the new ordering necessary for the other changes?